serverLib.sml

```diff
@@ -666,21 +675,39 @@ in
       val typ = read_job_type inp
                 handle IO.Io _ => cgi_die 500 ["cannot open ",f]
                      | Option => cgi_die 500 [f," has invalid file format"]
+      val worker = read_job_worker inp
       val format_type = if q = "finished" then span (status_attrs (read_status inp)) else String.concat
-      val last_date = if q = "running" then read_last_date inp else NONE
       val () = TextIO.closeIn inp
+      val inp2 = TextIO.openIn f
+      val {bcml,bhol} = read_bare_snapshot inp2
+             handle Option => cgi_die 500 [f," has invalid file format"]
+      val (head_sha,base_sha) =
+          case bcml of Bbr sha => (sha,sha)
+                     | Bpr {head_sha,base_sha} => (head_sha,base_sha)
+      val last_date = read_last_date inp2
       val ago_string =
-        case last_date of NONE => ""
+        case last_date of NONE => "No date"
         | SOME date => time_ago date
     in
-      span [("class","nowrap")]
-        [a (String.concat[base_url,"/job/",jid]) jid, " ",
-         format_type ["(", typ, ")",ago_string]]
+        [ a (String.concat[base_url,"/job/",jid]) jid
+        , format_type [typ]
+        , ago_string
+        , a (String.concat[cakeml_github,"/commit/",head_sha]) (code (String.substring (head_sha,0,7)))
+        , a (String.concat[cakeml_github,"/commit/",base_sha]) (code (String.substring (base_sha,0,7)))
+        , a (String.concat[hol_github,"/commit/",bhol])        (code (String.substring (bhol,0,7)))
+        , code (until_space worker)
+        ]
```

Comment:
Doesn't `inp2` need to be closed at some point?

Reply:
yep, fixed that